src/day3.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <vector>
+#include <fstream>
+
+std::vector<std::string> readPuzzleInputFromFile(std::string fileName) {
+    std::ifstream inFile(fileName);
+    std::string line;
+    std::vector<std::string> linesInFile{};
+
+    while (getline(inFile, line)) {
+        linesInFile.push_back(line);
+    }
+    return linesInFile;
+}
+
+int numberOfTreesOnSlope(const std::vector<std::string>& area, std::vector<int> slope){
```

Comment:
Instead of using a `std::vector<int>`, I think it's nicer and more readable to define a struct to do it, e.g.:

```
struct Slope
{
    int dx;
    int dy
};
```

Using things  like `slope.dx` is a lot more readable than `slope[0]`. Also note that since later C++-version (I'm not sure, I think 11 or 14), you can construct such structs as follows:

```
Slope slope{1, 3};
```

This even means that on line 51, you don't even have to change anything! This will work:

```
    const std::vector<Slope> slopesPart2 = { // slopes for part 2 {step right, step down}
        {1, 1},
        {3, 1},
        {5, 1},
        {7, 1},
        {1, 2}
    };
```